spec/unit/type/mailalias_spec.rb

```diff
@@ -4,13 +4,19 @@
 describe Puppet::Type.type(:mailalias) do
   include PuppetSpec::Files
 
+  if Puppet.features.microsoft_windows?
```

Comment:
Puppet already has helpers for temporary files and directories that abstract these details (and keep track of the generated paths to clean up later):

See the definition of `tmpfile` here:
https://github.com/puppetlabs/puppet/blob/c157bc9c2ee6422457bcb2bda70eadbf465c7c6c/spec/lib/puppet_spec/files.rb#L27

Grep on `tmpfile` and you'll see it's usage all over the place in the codebase.